pkg/validation/user.go

```diff
@@ -34,10 +34,6 @@ func ValidateUser(u *kubermaticv1.User) field.ErrorList {
 		allErrs = append(allErrs, field.Required(specPath.Child("email"), "this field is required"))
 	}
 
-	if u.Spec.ID == "" {
```

Comment:
check also down below in ValidateUserUpdate, ID is used, could you remove that as well?